modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/descriptor/BundleDescriptorUtils.java

```diff
@@ -36,33 +37,24 @@ public static boolean isCompatibleVersion(String availableVersion, String expect
       return true;
     }
 
-    Version available = getBundleVersion(availableVersion);
-    Version expected = getBundleVersion(expectedVersion);
+    Semver available = getBundleVersion(availableVersion);
```

Comment:
This is for plugins not for Mule artifacts. On plugins is where we support semantic versioning.